File: Interviews/LeetCode/Daily-Problems/2022/October/Day-14/2095.cpp
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        if (!head->next) return NULL;

        ListNode *slow = head, *fast = head->next->next;
        while (fast and fast->next) {
            slow = slow->next;
            fast = fast->next->next;
        }

        slow->next = slow->next->next;
        return head;
    }
};

File: Interviews/LeetCode/Daily-Problems/2022/October/Day-14/2095.py
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head.next: return None
        slow, fast = head, head.next.next
        while fast and fast.next:
            slow, fast = slow.next, fast.next.next
        
        slow.next = slow.next.next
        return head